File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.cc
#include <cstdio>

int main() {
    int n, m;
    int arr[100000];
    
    scanf("%d%d", &n, &m);
    
    for(int i=0; i<n; i++) {
        scanf("%d", arr+i);
        if(i > 0)
            arr[i] += arr[i-1];
    }
    
    for(int i=0; i<m; i++) {
        int a, b;
        
        scanf("%d%d", &a, &b);
        
        if(a != 1)
            printf("%d\n", arr[b-1]-arr[a-2]);
        else
            printf("%d\n", arr[b-1]);
    }
    
    return 0;
}

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
	
	static int n, m;
	static int[] arr;
	static StringBuilder sb = new StringBuilder();
	
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		n = Integer.parseInt(st.nextToken());
		m = Integer.parseInt(st.nextToken());
		arr = new int[n+1];
		
		st = new StringTokenizer(br.readLine());
		for(int i=1; i<=n; i++) {
			arr[i] = arr[i-1] + Integer.parseInt(st.nextToken());
		}
		
		for(int i=0; i<m; i++) {
			st = new StringTokenizer(br.readLine());
			int a = Integer.parseInt(st.nextToken());
			int b = Integer.parseInt(st.nextToken());
			
			sb.append(arr[b] - arr[a-1]).append("\n");
		}
		
		System.out.println(sb);
	}
}
